.github/workflows/aks.yaml

```diff
@@ -35,64 +36,103 @@ jobs:
+            --tags usage=${{ github.repository_owner }}-${{ github.event.repository.name }} owner=${{ steps.vars.outputs.owner }}
           az aks create \
-            --resource-group cilium-ci \
-            --name ${{ env.clusterName }} \
-            --location westeurope \
-            --node-count 2 \
+            --resource-group ${{ env.name }} \
+            --name ${{ env.name }} \
+            --location ${{ env.location }} \
             --network-plugin azure \
+            --node-count 2 \
+            --node-vm-size Standard_B2s \
+            --node-osdisk-size 30 \
+            --load-balancer-sku basic \
             --generate-ssh-keys
 
-      - name: Get Credentials
+      - name: Get cluster credentials
         run: |
-          az aks get-credentials --name ${{ env.clusterName }} --resource-group cilium-ci
+          az aks get-credentials \
+            --resource-group ${{ env.name }} \
+            --name ${{ env.name }}
 
-      - name: Install cilium
+      - name: Install Cilium
         run: |
-          cilium install --cluster-name ${{ env.clusterName }} --azure-resource-group cilium-ci --azure-tenant-id ${{ secrets.AZURE_TENANT_ID}} --azure-client-id ${{ secrets.AZURE_CLIENT_ID }} --azure-client-secret ${{ secrets.AZURE_CLIENT_SECRET }} --config monitor-aggregation=none
-
-      - name: Enable relay
+          cilium install \
+            --cluster-name=${{ env.name }} \
+            --azure-resource-group ${{ env.name }} \
+            --azure-tenant-id ${{ secrets.AZURE_TENANT_ID }} \
+            --azure-client-id ${{ secrets.AZURE_CLIENT_ID }} \
+            --azure-client-secret ${{ secrets.AZURE_CLIENT_SECRET }} \
+            --wait=false \
+            --config monitor-aggregation=none
+
+      - name: Enable Relay
         run: |
           cilium hubble enable
 
-      - name: Status
+      - name: Wait for Cilium status to be ready
         run: |
           cilium status --wait
 
-      - name: Relay Port Forward
+      - name: Port forward Relay
         run: |
           cilium hubble port-forward&
           sleep 5s
 
-      - name: Connectivity Test
+      - name: Run connectivity test
         run: |
           cilium connectivity test --all-flows
 
-      - name: Cleanup
-        if: ${{ always() }}
+      - name: Post-test information gathering
+        if: ${{ failure() }}
         run: |
           cilium status
           kubectl get pods --all-namespaces -o wide
           curl -sLO https://github.com/cilium/cilium-sysdump/releases/latest/download/cilium-sysdump.zip
           python cilium-sysdump.zip --output cilium-sysdump-out
-          az aks delete --yes --name ${{ env.clusterName }} --resource-group cilium-ci --no-wait
         shell: bash {0}
 
-      - name: Upload Artifacts
+      - name: Clean up AKS
+        if: ${{ always() }}
+        run: |
+          az group delete --name ${{ env.name }} --yes --no-wait
+        shell: bash {0}
```

Comment:
Is this line needed? We don't have the same on the cilium/cilium side.

Reply:
It is not needed in this specific instance, but I have decided to include it (and will also include on `cilium/cilium`) as I think it is warranted for this specific type of step.

In the case of the multicluster or external workloads workflows for example, this ensures even if the first cleanup fails that the second one will be run anyway.

We might in the future have additional resources that need to be cleaned up and that will get added to the cleanup steps in other workflows. Already having the `shell: bash {0}` ensures that we won't forget about adding it if that happens. Also, it keeps the overall workflow structure consistent :)

Reply:
> In the case of the multicluster or external workloads workflows for example, this ensures even if the first cleanup fails that the second one will be run anyway.

Could you add a comment to explain that? I don't really mind including it in all cleanup steps even if it's unnecessary, but that also means you'll have to copy the comment everywhere ;-)

Reply:
What do you mean, adding a comment?

Reply:
I mean adding a comment above `shell: bash {0}` to explain why it's sometimes needed and why we included it. So that others don't have to guess or ask you :-)

Reply:
Ah, I see. Is that really something we want to do...?
We were already leveraging custom shells in the post-test information gathering steps. It just disables the default fail-fast behaviour: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#exit-codes-and-error-action-preference
Since this is part of the default GH syntax and not a hidden trick / workaround, I would personally vote against commenting to explain why we have it, especially if we'd have to copy/paste it everywhere. It does not make sense to me, just like I would not comment `runs-on` or anything that's part of the default workflow syntax.

Reply:
What the line does was not obvious to me from (maybe it is to others), but on top of that, it doesn't say why it's needed for one step and not all of them.

If you used e.g. `.strategy.fail-fast` for one workflow and not all of them, I would expect a similar comment to explain the rationale. Since we're using `shell: bash {0}` for one step but not others, I think the same apply. The comment can be as simple as `Disable fail-fast to always try to delete all resources`.

Reply:
I think the difference comes from the fact you think this is documenting a design choice, whereas I think this is documenting basic GH syntax (e.g. it would be like if we documented our `id` and `steps.<id>.outputs` usage), because to me having a "fail-fast is disabled" marker is equivalent to having "always try to delete all resources".

But I get why it might be non-obvious, so I'll add a comment.

Reply:
Fixed in latest push :)